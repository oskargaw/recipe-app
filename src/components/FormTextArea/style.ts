import styled from "styled-components";
import { colors } from "../../theme";

export const StyledFormTextArea = styled.textarea`
  margin: 1rem 0;
  height: 2rem;
  border: none;
  border-radius: 0.3rem;
  background-color: ${colors.gray[100]};
  padding: 0.5rem 1rem;
  outline: none;
  font-family: sans-serif;
`;
