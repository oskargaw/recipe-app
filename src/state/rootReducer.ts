import { combineReducers } from "redux";
import { recipesReducer } from "./recipes/recipes.reducer";

const rootReducer = combineReducers({
  recipes: recipesReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

export default rootReducer;
