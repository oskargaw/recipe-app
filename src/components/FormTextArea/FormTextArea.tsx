import { UseFormRegister } from "react-hook-form";
import { FormLabel } from "../FormLabel";
import { StyledFormTextArea } from "./style";

export interface FormTextAreaProps {
  name: string;
  register: UseFormRegister<any>;
  label?: string;
  placeholder?: string;
}

export const FormTextArea: React.FC<FormTextAreaProps> = ({
  name,
  label,
  placeholder,
  register,
}) => (
  <>
    <FormLabel htmlFor={name}>{label}</FormLabel>
    <StyledFormTextArea {...register(name)} placeholder={placeholder} />
  </>
);
