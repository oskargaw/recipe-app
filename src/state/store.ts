import { createStore } from "redux";
import { persistReducer, persistStore } from "redux-persist"; // imports from redux-persist
import storage from "redux-persist/lib/storage";
import rootReducer from "./rootReducer";

const persistConfig = {
  key: "root",
  storage,
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = createStore(
  persistedReducer,
  (window as any).__REDUX_DEVTOOLS_EXTENSION__ &&
    (window as any).__REDUX_DEVTOOLS_EXTENSION__()
);

const persistor = persistStore(store as any);

export { store, persistor };
