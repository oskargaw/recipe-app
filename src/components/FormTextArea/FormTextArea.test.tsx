import { fireEvent, render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { RootState } from "../../state/rootReducer";
import { FormTextArea, FormTextAreaProps } from "./FormTextArea";

describe("<FormTextArea />", () => {
  const mockStore = configureStore([]);
  const initialState: RootState = {
    recipes: {
      recipeList: [],
    },
  };
  const store = mockStore(initialState);

  const mockRegister = jest.fn();

  const props: FormTextAreaProps = {
    name: "test",
    register: mockRegister,
    label: "test",
    placeholder: "test",
  };

  it("should render correctly", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <FormTextArea {...props} />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should display correct value after entering it inside the textarea", async () => {
    render(
      <Provider store={store}>
        <FormTextArea {...props} />
      </Provider>
    );

    const input = screen.getByPlaceholderText("test");
    expect(input).toBeInTheDocument();

    fireEvent.change(input, { target: { value: "4 bananas,brown sugar" } });

    expect(input).toHaveValue("4 bananas,brown sugar");
  });
});
