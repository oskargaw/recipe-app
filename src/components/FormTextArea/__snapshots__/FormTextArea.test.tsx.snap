// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormTextArea /> should render correctly 1`] = `
<DocumentFragment>
  <label
    class="sc-bdvvtL gJVqff"
    for="test"
  >
    test
  </label>
  <textarea
    class="sc-gsDKAQ OtYJG"
    placeholder="test"
  />
</DocumentFragment>
`;
